api/src/main/java/org/openmrs/module/referencedemodata/patient/DemoPatientGenerator.java

```diff
@@ -407,6 +439,79 @@ private void createDemoLabObs(Patient patient, Encounter encounter, Location loc
 		}
 	}
 	
+	private void createDemoAppointment(Patient patient, Date visitTime) {
+		Appointment appointment = new Appointment();
+        appointment.setPatient(patient);
+        Date startDateTime = toDate(toLocalDateTime(visitTime).plusMinutes(randomBetween(0, 60)));
+        Date endDateTime = toDate(toLocalDateTime(visitTime).plusMinutes(randomBetween(60, 90)));
+        appointment.setStartDateTime(startDateTime);
+        appointment.setEndDateTime(endDateTime);
+        appointment.setAppointmentKind(AppointmentKind.Scheduled);
+        
+        appointment.setService(getAppointmentServiceDefinitionService().getAllAppointmentServices(false).get(randomBetween(0, 20) % 2));
+        appointment.setAppointmentAudits(new HashSet<>());
+        Set<AppointmentProvider> appointmentProviders = new HashSet<>();
+        AppointmentProvider appointmentProvider = new AppointmentProvider();
+        appointmentProvider.setAppointment(appointment);
+        appointmentProvider.setProvider(getProviderService().getProviderByIdentifier("admin"));
+        appointmentProvider.setResponse(AppointmentProviderResponse.ACCEPTED);
+
+        appointmentProviders.add(appointmentProvider);
+        appointment.setProviders(appointmentProviders);
+        Context.addProxyPrivilege(PrivilegeConstants.MANAGE_APPOINTMENTS);
+        getAppointmentsService().validateAndSave(appointment);
+        Context.removeProxyPrivilege(PrivilegeConstants.MANAGE_APPOINTMENTS);
+        
+	}
+	
+	private void createDemoPatientProgram(Patient patient, Program program, Date visitTime) {
+		
+		Date startDate = toDate(toLocalDateTime(visitTime).plusHours((randomBetween(0, 72))));
+		ProgramWorkflowState state1 = program.getAllWorkflows().stream().findFirst().get().getStates().stream().findFirst().get();
```

Comment:
`getAllWorkflows()` potentially returns retired workflows.

```suggestion
		ProgramWorkflowState state1 = program.getWorkflows().stream().findFirst().get().getStates().stream().findFirst().get();
```

Reply:
What happens if there are programs with:

1. No workflows defined?
2. A workflow defined with no states?

Reply:
Patient program should be saved without a workflow and the same should happen if states where not configured for a program workflow. I've added a changes to this effect.